test/core.js

```diff
@@ -67,6 +67,16 @@ test( "attribute selectors with naked '#'", function() {
 			jQuery( "a[href=\"#junk\"]" );
 		} catch ( e ) {}
 	} );
+
+	expectNoWarning( "ready function with attribute selector", function() {
+		try {
+			jQuery( function() {
+				if ( jQuery.thisIsNeverTrue ) {
+					jQuery( "a[href=#]" );
```

Comment:
Because the is-it-a-string check was missing, the old code takes the function and stringifies the body during `rattrHash.test( selector )`. So if the stringified function had a pattern that looks like a bad one, it would try to give a warning when it should not--it should only give the warning when the code with the selector actually runs and we shouldn't even be checking functions here. Since the document is ready the function *will* run immediately and the `if` just  makes sure the selector isn't actually executed.

So yeah. :smile: I'm really surprised this didn't cause trouble before but I guess JavaScript will pretty much stringify anything.